scipy/stats/stats.py

```diff
@@ -3123,17 +3124,56 @@ def trim_mean(a, proportiontocut, axis=0):
 
 
 class F_onewayConstantInputWarning(RuntimeWarning):
-    """Warning generated by `f_oneway` when an input is constant, e.g.
-       each of the samples provided is a constant array"""
+    """
+    Warning generated by `f_oneway` when an input is constant, e.g.
+    each of the samples provided is a constant array.
+    """
 
     def __init__(self, msg=None):
         if msg is None:
             msg = ("Each of the input arrays is constant;"
-                    "the F-value is not defined or infinite")
+                   "the F statistic is not defined or infinite")
         self.args = (msg,)
 
 
-def f_oneway(*args):
+class F_onewayBadInputSizesWarning(RuntimeWarning):
+    """
+    Warning generated by `f_oneway` when an input has length 0,
+    or if all the inputs have length 1.
+    """
+    pass
+
+
+def _create_f_oneway_nan_result(shape, axis):
+    """
+    This is a helper function for f_oneway for creating the return values
+    in certain degenerate conditions.  It creates return values that are
+    all nan with the appropriate shape for the given `shape` and `axis`.
+    """
+    axis = np.core.multiarray.normalize_axis_index(axis, len(shape))
+    shp = shape[:axis] + shape[axis+1:]
+    if shp == ():
+        f = np.nan
+        prob = np.nan
+    else:
+        f = np.full(shp, fill_value=np.nan)
+        prob = f.copy()
+    return F_onewayResult(f, prob)
+
+
+def _first(arr, axis):
```

Comment:
not sure but this might be used more generally? future PR could move it to `_lib` if you agree (and find a use outside of this) but doesn't need to happen in this PR. But maybe is not worth it as your comment indicates is to be removed once minimal supported numpy is large enough.